lib/gtfs.ex

```diff
@@ -185,6 +190,31 @@ defmodule Gtfs do
     end
   end
 
+  @spec retrive_data(files(), boolean) :: t()
+  defp retrive_data(files, mocked_files?) do
+    if !mocked_files? && CacheFile.should_use_file?() do
+      Logger.info("Loading gfts data from cached file")
+
+      case CacheFile.load_gtfs() do
+        {:ok, state} ->
+          Logger.info("Loaded gtfs data from cached file")
+          state
+
+        _ ->
+          Logger.info("Loading gtfs data remote files")
+          state = parse_files(files)
+          Logger.info("Loaded gtfs data remote files")
+          CacheFile.save_gtfs(state)
+          state
+      end
+    else
+      Logger.info("Loading gtfs data remote files")
+      state = parse_files(files)
+      Logger.info("Loaded gtfs data remote files")
+      state
+    end
+  end
```

Comment:
If we use the cache, we end up downloading the files, and then discarding them. What do you think about calling the cache in the fetch step instead of the parse step??

It'd require moving the call to `parse_files` into that function, but I think that's fine. Something like (very roughly)
```elixir
@spec fetch_gtfs(files_source()) :: t() | {:error, any()}
defp fetch_gtfs({:url, url}) do
  if should_cache and cache_exists do
    Cache.load()
  else
    zip_binary = HTTPoison.get()
    zip_binary |> unzip() |> parse()

defp fetch_gtfs({:mocked_files}, mocked_files) do
  mocked_files |> parse()
```

Small benefit: it's easier to not cache the mocked_files, since the cache would only be looked at in the url case.

Reply:
This was a good catch. The latest commit does this - it's a bit hairy to read through, but it seems to work.

Plus the time to load from a cache drops even further to ~80ms!